### DISCUSSION TITLE
api support encryption ?

### DISCUSSION THREAD

**Discussion by ghost** [directus/directus#13195] [Feature Requests]
Is it possible to encrypt the api? Can enhance data security
For example, using rsa+aes encryption

**Comment by rijkvanzanten** [comment]
Do you have any examples of what that might look like in practice? I'm not quite sure what you mean by "encrypt the api"

**Reply by ghost** [reply]
https://blog.csdn.net/weixin_46560512/article/details/123551153
https://blog.csdn.net/qq_40323256/article/details/124017789

step 1:After the client gets the decrypted key, it starts to request the interface data
step 2:The backend encrypts the json data to be encrypted and returns it to the client
step 3:The client gets the encrypted json data for decryption

**Reply by azrikahar** [reply]
There's no out-of-the-box support for this per se, but you may be able to achieve this by adding a middleware (using custom extensions with Init event: https://docs.directus.io/extensions/hooks/#init-events) such as [express-crypto](https://www.npmjs.com/package/express-crypto) or [my-encrypter-middleware](https://www.npmjs.com/package/my-encrypter-middleware) (just some examples I found from a brief search).

**Reply by ghost** [reply]
Thank you very much for your answer, and thank you for providing such an excellent open source project
